scripts/install-nix-from-closure.sh

```diff
@@ -1,149 +1,618 @@
-        chmod -w "$dest/store/$i"
-    fi
-done
-echo "" >&2
-
-echo "initialising Nix database..." >&2
-if ! $nix/bin/nix-store --init; then
-    echo "$0: failed to initialize the Nix database" >&2
-    exit 1
-fi
-
-if ! "$nix/bin/nix-store" --load-db < "$self/.reginfo"; then
-    echo "$0: unable to register valid paths" >&2
-    exit 1
-fi
-
-. "$nix/etc/profile.d/nix.sh"
-
-if ! "$nix/bin/nix-env" -i "$nix"; then
-    echo "$0: unable to install Nix into your default profile" >&2
-    exit 1
-fi
-
-# Install an SSL certificate bundle.
-if [ -z "$NIX_SSL_CERT_FILE" ] || ! [ -f "$NIX_SSL_CERT_FILE" ]; then
-    $nix/bin/nix-env -i "$cacert"
-    export NIX_SSL_CERT_FILE="$HOME/.nix-profile/etc/ssl/certs/ca-bundle.crt"
-fi
-
-# Subscribe the user to the Nixpkgs channel and fetch it.
-if ! $nix/bin/nix-channel --list | grep -q "^nixpkgs "; then
-    $nix/bin/nix-channel --add https://nixos.org/channels/nixpkgs-unstable
-fi
-if [ -z "$_NIX_INSTALLER_TEST" ]; then
-    $nix/bin/nix-channel --update nixpkgs
-fi
-
-added=
-if [ -z "$NIX_INSTALLER_NO_MODIFY_PROFILE" ]; then
-
-    # Make the shell source nix.sh during login.
-    p=$HOME/.nix-profile/etc/profile.d/nix.sh
-
-    for i in .bash_profile .bash_login .profile; do
-        fn="$HOME/$i"
-        if [ -w "$fn" ]; then
-            if ! grep -q "$p" "$fn"; then
-                echo "modifying $fn..." >&2
-                echo "if [ -e $p ]; then . $p; fi # added by Nix installer" >> "$fn"
+
+    if [ -d "$dest" ]; then
+        print "Directory $dest already exists. Skipping creation of $dest"
+    fi
+
+}
+
+installNix() {
+
+    # In this function, if some action makes any change to the system, or does err Exit - it's need to be wrapped in a function, ex func(){...}. Inside func() you start with setup of traps, before errors/actions happen, so and according revert function be launched: 'trap funcRevertOnCondition condition', and funcRevertOnCondition is described right after trap.
+    # How all actions operate inside this function:
+    #**************************************************
+    #
+    #    installNix() {
+    #
+    #        func() {
+    #
+    #            trap 'funcRevertOnCondition $LINENO $?' condition
+    #            funcRevertOnCondition() {
+    #                errorRevert "Received Error signal from line $1"
+    #                # revert action
+    #                >&2 contactme # This function gives contact information
+    #                exit "$?"
+    #            }
+    #
+    #            # action of functions
+    #            # 'exit SING # to the trap that catches exit SIGN'
+    #        }
+    #
+    #        func
+    #
+    #    }
+    #
+    #**************************************************
+
+    # Function must have revert procedure. In case of any problem - function quits without artifacts/changes to the system
+    # Nix is pure, and claims to be transnational, so let's show people clean transnational installation
```

Comment:
transactional?

Reply:
Yeah. I watched for the word `transactional` on the main page, and where it gone.

I remember discussion I held with the person, and Nix stated that it's transactional.
Now I think maybe I was dreaming.

Reply:
It's for sure stateless, and putting changes in place is atomic, so you could say it's transactional.

Reply:
Yes.
Fixed. In: https://github.com/NixOS/nix/pull/1565/commits/8dbed04927df23f2d9359c81685bd15d3b63885f